src/devices/src/virtio/block/device.rs

```diff
@@ -62,11 +67,29 @@ impl DiskProperties {
         is_disk_read_only: bool,
         cache_type: CacheType,
     ) -> io::Result<Self> {
-        let mut disk_image = OpenOptions::new()
+        let disk_image = OpenOptions::new()
```

Comment:
It is vital that all accesses to a single disk image go through a single object (write accesses at least), or you risk data corruption because of the internal metadata cache.

So as I suggested before, I think this should be an `Arc<>`; the image should be opened in `Block::new()`, the `Arc<SyncFormatAccess<ImagoFile>>` kept in `Block` (replacing the `disk_image_path`), and consequently also the image ID needs to be built there, then. Something like [0001-Keep-single-instance-of-SyncFormatAccess.txt](https://github.com/user-attachments/files/17734560/0001-Keep-single-instance-of-SyncFormatAccess.txt) should work, I believe. (Sorry for it being .txt instead of .patch, github seems to hate .patch files from Linux users for some reason[[1]](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/attaching-files).)

Reply:
Should be fixed now